map_creator/src/load_reachability_map.cpp

```diff
@@ -0,0 +1,235 @@
+    chunk_size=(dims_out[0]/10);
+    offset[0] = 0;
+    
+    for(int it=0;it<chunk_itr;it++)
+    {
+      
+      offset[1] = 0;
+	 
+      
+      if ((dims_out[0]-(chunk_size*it))/chunk_size != 0){
+	count[0] = chunk_size;
+	offset[0] = chunk_size*it;
+	
+      }
+      else{
+	count[0] = (dims_out[0]-(chunk_size*it));
+	offset[0] = count[0];
+      }
+      
+      
+      count[1]  = 10;
+      
+      double data_out[count[0]][count[1]];
+	
+      status = H5Sselect_hyperslab (dataspace, H5S_SELECT_SET, offset, NULL, 
+                                  count, NULL);
+	hsize_t dimsm[2]; 
+	dimsm[0] = count[0];
+    	dimsm[1] =  count[1];
+	hid_t memspace;
+	
+        memspace = H5Screate_simple (RANK_OUT, dimsm, NULL);  
+
+	status = H5Dread (dataset, H5T_NATIVE_DOUBLE, memspace, dataspace,
+                      H5P_DEFAULT, data_out);
+
+	
+	for(int i=0;i<count[0];i++){
+
+          vector<double> sphereCenter;
+	  vector<double> Poses;
+ 	  for(int j=0;j<3;j++)
+	  {
+		 sphereCenter.push_back(data_out[i][j]);
+		 }
+	  
+	  for(int k=3;k<10;k++)
+	  {
+		 Poses.push_back(data_out[i][k]);
+		}
+	    
+	PoseColFilter.insert(pair<vector<double>, vector<double> >(sphereCenter, Poses));
+	}
+  }
+
+}
+
+void h5ToMultiMapSpheres(const hid_t dataset, multimap<vector<double>, double >& SphereCol){
+
+    hsize_t dims_out[2], count[2], offset[2], dimsm[2];
+    hid_t dataspace = H5Dget_space (dataset);    /* dataspace handle */
+    int rank      = H5Sget_simple_extent_ndims (dataspace);
+    herr_t status_n  = H5Sget_simple_extent_dims (dataspace, dims_out, NULL);
+    herr_t status;
+    offset[0] = 0;
+    offset[1] = 0;
+    count[0]  = dims_out[0];
+    count[1]  = 4;
+    double data_out[count[0]][count[1]];
+    status = H5Sselect_hyperslab (dataspace, H5S_SELECT_SET, offset, NULL, 
+                                  count, NULL);
+    dimsm[0] = count[0];
+    dimsm[1] =  count[1];
+    hid_t memspace;
+    memspace = H5Screate_simple (RANK_OUT, dimsm, NULL);  
+    status = H5Dread (dataset, H5T_NATIVE_DOUBLE, memspace, dataspace,
+                      H5P_DEFAULT, data_out);
+    
+    for(int i=0;i<count[0];i++){
+          vector<double> sphereCenter;
+	  double ri;
+ 	  for(int j=0;j<3;j++)
+	    {
+		 sphereCenter.push_back(data_out[i][j]);
+	    }
+	  
+	  for(int k=3;k<4;k++)
+	    {
+		 ri = data_out[i][k];
+	    }
+	    
+	  SphereCol.insert(pair<vector<double>, double >(sphereCenter, ri));
+	}
+}
+
+};
+
+
+int main(int argc, char **argv)
+{
+    if (argc<2){
+	ROS_ERROR_STREAM("Please provide the name of the reachability map. If you have not created it yet, Please create the map by running the create reachability map launch file in map_creator");
+	return 1;
+    } 
+    else{
+	
+    ros::init(argc, argv, "workspace");
+    ros::NodeHandle n;
+    
+
+    ros::Publisher workspace_pub = n.advertise<map_creator::WorkSpace>("reachability_map", 1);
+    ros::Rate loop_rate(10);
+  
+    int count = 0;
+    while (ros::ok())
```

Comment:
Same comment about while loop.  Use a latched topic if possible to keep while loop simple and minimize computations.